nilearn/plotting/html_surface.py

```diff
@@ -406,22 +581,43 @@ def view_surf(surf_mesh, surf_map=None, bg_map=None, threshold=None,
     nilearn.plotting.view_img_on_surf: Surface plot from a 3D statistical map.
 
     """
-    surf_mesh = surface.load_surf_mesh(surf_mesh)
-    if surf_map is None:
-        surf_map = np.ones(len(surf_mesh[0]))
-    else:
-        surf_mesh, surf_map = surface.check_mesh_and_data(
-            surf_mesh, surf_map)
-    if bg_map is not None:
-        _, bg_map = surface.check_mesh_and_data(surf_mesh, bg_map)
-    info = one_mesh_info(
-        surf_map=surf_map, surf_mesh=surf_mesh, threshold=threshold,
-        cmap=cmap, black_bg=black_bg, bg_map=bg_map,
-        bg_on_data=bg_on_data, darkness=darkness,
-        symmetric_cmap=symmetric_cmap, vmax=vmax, vmin=vmin)
-    info['colorbar'] = colorbar
-    info['cbar_height'] = colorbar_height
-    info['cbar_fontsize'] = colorbar_fontsize
-    info['title'] = title
-    info['title_fontsize'] = title_fontsize
-    return _fill_html_template(info, embed_js=True)
+    _check_engine(engine)
+    if engine == "plotly":
+        surf_mesh = surface.load_surf_mesh(surf_mesh)
+        if surf_map is None:
+            surf_map = np.ones(len(surf_mesh[0]))
+        else:
+            surf_mesh, surf_map = surface.check_mesh_and_data(
+                surf_mesh, surf_map)
+        if bg_map is not None:
+            _, bg_map = surface.check_mesh_and_data(surf_mesh, bg_map)
```

Comment:
This code is duplcated below. You can move it out of and above the if statement and then delete the duplicate in the elif statement